lib/apaka/packaging/autoproj2adaptor.rb

```diff
@@ -545,15 +545,22 @@ def dependencies(pkg, with_rock_release_prefix = true)
                     # Convert native ruby gems package names to rock-xxx
                     if pkg_handler == "gem"
                         pkg_list.each do |name|
-                            version = nil
-                            if name =~ /([<>=]=?.*)$/
-                                version = $1
-                            end
+                            if name.class == Hash
+                                git = name['git']
+                                name = name['name']
+                                version = nil
```

Comment:
would suggest, that you allow version to be set if available, e.g.
version = name['version'] if name.has_key?(version)